iavl/iavl_tree.go

```diff
@@ -1,37 +1,131 @@
+/*
+	Package merkle implements a key/value store with an immutable Merkle tree.
+*/
+
 package iavl
 
 import (
 	"bytes"
 	"container/list"
+	"fmt"
 	"sync"
 
 	wire "github.com/tendermint/go-wire"
-	. "github.com/tendermint/tmlibs/common"
-	dbm "github.com/tendermint/tmlibs/db"
+	cmn "github.com/tendermint/tmlibs/common"
+	"github.com/tendermint/tmlibs/db"
 	"github.com/tendermint/tmlibs/merkle"
 )
 
+// Keys for administrative persistent data
+var rootsKey = []byte("go-merkle.roots")      // Database key for the list of roots
```

Comment:
Yes, I missed those prefixes :-) We'll stick with merkleeyes.xxxx? 

I'll see if I can wind all of the extra data into a single struct. I don't think I'll be able to get a nice consolidated load/save for the database, since I am accessing each key explicitly.